pkg/controller/kubemon/statefulset.go

```diff
@@ -0,0 +1,385 @@
+package kubemon
+
+import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/apis/dynatrace/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/customproperties"
+	"github.com/Dynatrace/dynatrace-operator/pkg/dtclient"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	MonitoringServiceAccount = "dynatrace-kubernetes-monitoring"
+	KubernetesArch           = "kubernetes.io/arch"
+	KubernetesOs             = "kubernetes.io/os"
+	KubernetesBetaArch       = "beta.kubernetes.io/arch"
+	KubernetesBetaOs         = "beta.kubernetes.io/os"
+
+	AMD64 = "amd64"
+	ARM64 = "arm64"
+	LINUX = "linux"
+
+	DtTenant          = "DT_TENANT"
+	DtServer          = "DT_SERVER"
+	DtToken           = "DT_TOKEN"
+	DtCapabilities    = "DT_CAPABILITIES"
+	DtIdSeedNamespace = "DT_ID_SEED_NAMESPACE"
+	DtIdSeedClusterId = "DT_ID_SEED_K8S_CLUSTER_ID"
+
+	DtTenantArg       = "--tenant=$(DT_TENANT)"
+	DtTokenArg        = "--token=$(DT_TOKEN)"
+	DtServerArg       = "--server=$(DT_SERVER)"
+	DtCapabilitiesArg = "--enable=$(DT_CAPABILITIES)"
+
+	ProxyArg = `PROXY="${ACTIVE_GATE_PROXY}"`
+	ProxyEnv = "ACTIVE_GATE_PROXY"
+	ProxyKey = "ProxyKey"
+
+	CapabilityEnv = "kubernetes_monitoring"
+)
+
+func newStatefulSet(instance v1alpha1.DynaKube, tenantInfo *dtclient.TenantInfo, kubeSystemUID types.UID) *v1.StatefulSet {
```

Comment:
Would suggest to create the whole object here.

Reply:
merged as many build* functions as I am comfortable with